### PR TITLE
Improvement/59 Make board work with context freezing

### PR DESCRIPTION
**Pull request by NeillM**
As part of this I split the ability to post and rate into a new capability that is separate from the view capability.

I added the new capability into the checks for users being able to edit various features

A also made it so that comments also required the ability for users to make a comment in the first place.

An example of the admin user viewing a board in frozen context with these changes applied:
image

and the admin user viewing a comment in a frozen board:
image

### CODE REVIEW

**Comment by NeillM** [comment]
This should resolve #59

**Comment by NeillM** [comment]
I have rebased my changes for this on onto the MOODLE_400_STABLE branch as well

File: classes/board.php [hunk 1]
```diff
@@ -1264,10 +1264,17 @@ public static function board_is_editor($boardid) {
...
      * @param int $boardid
-     * @return void
+     * @return boolean
      */
     public static function board_users_can_edit($boardid) {
         global $DB;
+
+        $context = static::context_for_board($boardid);
+        if (!has_capability('mod/board:post', $context)) {
+            // the user is not allowed to post via capabilities.
```
Comment (learningtechnologyservices) [inline classes/board.php]: Sorry for this request, but CI checks are god, and they prefer comments starting with capital letters. TIA!

File: version.php [hunk 2]
```diff
@@ -25,7 +25,7 @@
 defined('MOODLE_INTERNAL') || die;
 
 $plugin->component = 'mod_board'; // Full name of the plugin (used for diagnostics).
-$plugin->version   = 2022040105; // The current module version Use 2022.04.01 as base for 4.00.
+$plugin->version   = 2022040106; // The current module version Use 2022.04.01 as base for 4.00.
```
Comment (learningtechnologyservices) [inline version.php]: Version on main branch is now 2022040111, also release is '1.401.02 (Build 2022040111)';

Hoping to include this in the next release shortly, with some other PRs. TIA!

**Review by learningtechnologyservices — CHANGES_REQUESTED** [review]
Hi Neill, many thanks for submitting this PR for improving Board! We'd like to include this in the next release, if we can get these couple of edits done by yourself please? TIA! Karen

P.S. We're just maintaining one active branch, MOODLE_401_STABLE, thanks!

**Comment by NeillM** [comment]
I have rebased the changes (there was a merge conflict that was not part of the version file in the first commit)

I fixed the capitalisation on the comment as part of the rebase.

**Comment by danowar2k** [comment]
Eagerly awaiting the split of post and view capabilities :-)

**Comment by NeillM** [comment]
Rebased the change

**Review by learningtechnologyservices — APPROVED** [review]

**Pull request merged**